src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-attribute.mdx

```diff
@@ -0,0 +1,722 @@
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java]
+
+```java
+NewRelic.setAttribute(string $name, string or float $value)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+NewRelic.setAttribute(name: String?, value: String?)
+NewRelic.setAttribute(name: String?, value: Double)
+NewRelic.setAttribute(name: String?, value: Boolean)
+```
+
+## Description [#description]
+
+This static method creates a session-level custom [attribute](/docs/insights/explore-data/custom-events/insert-custom-attributes-events-new-relic-mobile#attribute-definition) with a specified name and value. This method overwrites its previous value and type each time it is called. The created attribute is shared by multiple [Mobile event types](/docs/insights/insights-data-sources/default-data/mobile-events).
+
+For more context on how to use this API, see the [Android API guide](/docs/mobile-monitoring/new-relic-mobile-android/install-configure/work-android-sdk-api#attributes-events-insights).
+
+You can override any of the [MobileSession default attributes](/docs/insights/insights-data-sources/default-attributes/mobile-default-attributes-insights) except:
+
+* `appId`
+* `appName`
+* `accountId`
+* `carrier`
+* `category`
+* `deviceManufacturer`
+* `deviceModel`
+* `eventType`
+* `install`
+* `lastInteraction`
+* `memUsageMb`
+* `newRelicVersion`
+* `osMajorVersion`
+* `osName`
+* `osVersion`
+* `platform`
+* `platformVersion`
+* `sessionDuration`
+* `sessionId`
+* `timestamp`
+* `type`
+* `upgradeFrom`
+* `uuid`
+* Anything prefixed with `NewRelic`
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$name`
+      </td>
+
+            <td>
+        `string`
+      </td>
+
+      <td>
+        Required. Name of the attribute.
+      </td>
+    </tr>
+
+        <tr>
+      <td>
+        `$value`
+      </td>
+
+            <td>
+        `string` or `float`
+      </td>
+
+      <td>
+        Required. The second parameter passed can be either a `string` value or a `float` value.
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if recorded successfully, and `false` if it doesn't.
+
+## Example [#example]
+
+Here's an example of creating an attribute with the specified text name and text value. `setAttribute` overwrites its previous value and type each time it is called:
+
+### Java [#java]
+
+```java
+boolean attributeSet = NewRelic.setAttribute("storeId", "SampleStoreId000");
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
+
+Here's an example of creating an attribute with the specified text name and numeric value:
+
+### Java [#java]
+
+```java
+boolean attributeSet = NewRelic.setAttribute("rate", 9999.99);
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
val attributeSet = NewRelic.setAttribute("rate", 9999.99)

Reply:
```suggestion
```kotlin
    val attributeSet = NewRelic.setAttribute("rate", 9999.99)
```
```